server/src/main/java/org/eclipse/openvsx/RegistryAPI.java

```diff
@@ -976,45 +1016,47 @@ public ResponseEntity<SearchResultJson> search(
             @RequestParam(required = false)
             @Parameter(description = "Whether to include information on all available versions for each returned entry")
             boolean includeAllVersions
-    ) {
-        if (size < 0) {
-            var json = SearchResultJson.error("The parameter 'size' must not be negative.");
-            return new ResponseEntity<>(json, HttpStatus.BAD_REQUEST);
-        }
-        if (offset < 0) {
-            var json = SearchResultJson.error("The parameter 'offset' must not be negative.");
-            return new ResponseEntity<>(json, HttpStatus.BAD_REQUEST);
-        }
+	) {
+		try (ScopeLog sl = new ScopeLog(LOGGER, Level.FINER, "/api/-/search")) {
```

Comment:
Using a special class for logging might not be a good idea. For OpenVSX to survive it will be necessary to introduce some Application Performance Monitoring in the future, with its own monitoring. Instead to adjust the code for new logging sinks, I would propose to instrument the code with some standard logging, such as OpenTelemetry. Take a look at their [server example](https://github.com/open-telemetry/opentelemetry-java-examples/blob/923a93e38d4f3b7ebf6f365de44550bbe9e77ecf/http/src/main/java/io/opentelemetry/example/http/HttpServer.java#L71). The code looks quite similar, but it is a standard API which could be listened by many log/trace implementations.

Reply:
This is a fair assessment. I see monitoring and postmortem tracing as two separate issues. What I would like in the future is to add OTEL to the scopelogs so that the loggers can feed monitoring.